Makefile

```diff
@@ -1,21 +1,77 @@
+SHELL := bash
+.ONESHELL:
+.SHELLFLAGS := -eu -o pipefail -c
+.DELETE_ON_ERROR:
+MAKEFLAGS += --warn-undefined-variables
+MAKEFLAGS += --no-builtin-rules
+MAKEFLAGS += --no-print-directory
+MAKEFLAGS += -S
+
 .DEFAULT: all
 
+CARGO ?= cargo
+CARGO_WATCH ?= cargo-watch
+CARGO_CLIPPY ?= cargo-clippy
+DOCKER ?= docker
+
 all: build test
 
-build:
-	cargo build --features cli
+help:  ## Display this help
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_\-.*]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)

```

Comment:
This is a `Makefile` formula for printing help text and is pretty common across all the wasmCloud projects, and is generally pretty helpful because it's a facility Makefiles lack. 

It'd be great if we could move to `Justfile`s in the future (it's built-in there), but I think not having a `help` target would be a bad for ergonomics.